jwst/assign_wcs/util.py

```diff
@@ -1420,3 +1420,16 @@ def update_fits_wcsinfo(datamodel, max_pix_error=0.01, degree=None, npoints=32,
     datamodel.meta.wcsinfo.instance.update(hdr_dict)
 
     return hdr
+
+
+def populate_wavelength_extension(model):
+    """For MIRI LRS Slitless, no step currently
+    assigns values to model.wavelength - do it here
+    """
+    data_shape = model.data.shape
+    if len(data_shape) > 2:
+        data_shape = data_shape[-2:]
+    index_array = np.indices(data_shape[::-1])
```

Comment:
I think the `x`, `y` inputs should be generated from the `bounding_box`, because I believe these need to be in full frame coordinates. The way to do this is
```
from gwcs.wcstools import grid_from_bounding_box

x, y = grid_from_bounding_box(data_model.wcs.bounding_box)
```

Reply:
This is the sort of thing I expected to miss - thanks for the heads up! I get a similar solution with the new code, but the coords did shift a small amount.

Reply:
`x`, and `y` should be integers (the center of the pixel). Is this what you get from `grid_from_bounding_box`?

Reply:
Yes, the return works as you expected - x is a 2d array of x-positions, while y is a 2d array of y-positions, such that x[i] and y[i] specify a given pixel. They may be of type float (I checked yesterday but can't remember now), but the values are ints.